import/index_java/src/org/vufind/index/MultiFormatCalculator.java

```diff
@@ -0,0 +1,448 @@
+package org.vufind.index;
+/**
+ * Alternate, multi-valued format determination logic.
+ *
+ * Copyright (C) Villanova University 2018.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+import org.marc4j.marc.Record;
+import org.marc4j.marc.ControlField;
+import org.marc4j.marc.DataField;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Alternate, multi-valued format determination logic.
+ */
+public class MultiFormatCalculator
+{
+    /**
+     * Determine whether a record cannot be a book due to findings in 007.
+     *
+     * @param char formatCode
+     * @return boolean
+     */
+    protected boolean definitelyNotBookBasedOn007(char formatCode) {
+        switch (formatCode) {
+            // Things that are not books: filmstrips/transparencies (G),
+            // pictures (K) and videos/films (M, V):
+            case 'G':
+            case 'K':
+            case 'M':
+            case 'V':
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Determine whether a record cannot be a book due to findings in leader
+     * and fixed fields.
+     *
+     * @param char formatCode
+     * @param ControlField fixedField
+     * @return boolean
+     */
+    protected boolean definitelyNotBookBasedOnRecordType(char recordType, ControlField fixedField) {
```

Comment:
fixedField is a confusing parameter name. Looks like it's 008, no?

Reply:
The 008 is commonly referred to as "fixed fields" (at least around here) because its official name in the spec is "Fixed-Length Data Elements." But I'll change it to reduce ambiguity!

Reply:
That explains! But also 006 is "Fixed-Length Data Elements[...]" and 007 is "Physical Description Fixed Field", so I was a bit confused since I first thought it might be all fixed-length fields (005-008).

Reply:
That's fair! In any case, thanks again for the feedback, and I'll finish up the more complex parts of the refactoring a little later on.